java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -13,227 +13,286 @@
  * Method:    createSession
  * Signature: (JJLjava/lang/String;J)J
  */
-JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OrtSession_createSession__JJLjava_lang_String_2J
-  (JNIEnv * jniEnv, jclass jclazz, jlong apiHandle, jlong envHandle, jstring modelPath, jlong optsHandle) {
-    (void) jclazz; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    OrtSession* session;
+JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OrtSession_createSession__JJLjava_lang_String_2J(JNIEnv* jniEnv, jclass jclazz, jlong apiHandle, jlong envHandle, jstring modelPath, jlong optsHandle) {
+  (void)jclazz;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  OrtSession* session = NULL;
 
 #ifdef _WIN32
-    const jchar* cPath = (*jniEnv)->GetStringChars(jniEnv, modelPath, NULL);
-    size_t stringLength = (*jniEnv)->GetStringLength(jniEnv, modelPath);
-    wchar_t* newString = (wchar_t*)calloc(stringLength + 1, sizeof(wchar_t));
-    if(newString == NULL) {
-        throwOrtException(jniEnv, 1, "Not enough memory");
-        return 0;
-    }
-    wcsncpy_s(newString, stringLength+1, (const wchar_t*) cPath, stringLength);
-    checkOrtStatus(jniEnv,api,api->CreateSession((OrtEnv*)envHandle, newString, (OrtSessionOptions*)optsHandle, &session));
-    free(newString);
-    (*jniEnv)->ReleaseStringChars(jniEnv,modelPath,cPath);
+  const jchar* cPath = (*jniEnv)->GetStringChars(jniEnv, modelPath, NULL);
+  size_t stringLength = (*jniEnv)->GetStringLength(jniEnv, modelPath);
+  wchar_t* newString = (wchar_t*)calloc(stringLength + 1, sizeof(wchar_t));
+  if (newString == NULL) {
+    (*jniEnv)->ReleaseStringChars(jniEnv, modelPath, cPath);
+    throwOrtException(jniEnv, 1, "Not enough memory");
+    return 0;
+  }
+  wcsncpy_s(newString, stringLength + 1, (const wchar_t*)cPath, stringLength);
```

Comment:
>wcsncpy_s(newString, stringLength + 1, (const wchar_t*)cPath, stringLength); [](http://example.com/codeflow?start=2&length=76)

So the only reason we make a copy is a zero terminator?

Reply:
Yeah, the java UTF-16 string is not null terminated, and I think the windows path loading logic requires that it be a null terminated string.